### PR TITLE
add limit to the rules api

### PR DESCRIPTION
**Pull request by Alphasaurs**
Trying to add limit and close this issue #9342
thanks @LeviHarrison for review I have incorporated your comments in this pr
please do let me know for other changes if any.

Signed-off-by: alphasaurs <alphasaurs@gmail.com>

### CODE REVIEW

**Comment by LeviHarrison** [comment]
Closing and re-opening to trigger CI.

**Review by LeviHarrison — COMMENTED** [review]
Almost there! Thanks for making those changes, and DCO looks good this time. Presuming the tests pass, which I think they will, there's only one more thing: the `limit` field should be added to the sample API response right below the `interval` field.

https://github.com/prometheus/prometheus/blob/129ed4ec8baf32f9bccd6a73508db45dbec9cbf5/docs/querying/api.md#L622

**Comment by Alphasaurs** [comment]
I have added that too do let me know for tasks that needed to be completed before this release would like to learn and contribute more to this project

File: docs/querying/api.md [hunk 1]
```diff
@@ -620,6 +620,7 @@ $ curl http://localhost:9090/api/v1/rules
                 ],
                 "file": "/rules.yaml",
                 "interval": 60,
+                "limit":    0,
```
Comment (LeviHarrison) [inline docs/querying/api.md:623]: nit: no extra spacing is needed here.

```suggestion
                "limit": 0,
```

**Comment by LeviHarrison** [comment]
> let me know for tasks that needed to be completed before this release would like to learn and contribute more to this project

As the release will be cut tomorrow, I don't think there's anything left to get it, but you can certainly contribute to future releases. Nothing specific comes to mind right now, but you can always find more tasks in the issue tracker. So glad for your enthusiasm in the project!

File: docs/querying/api.md (hunk 1 above)
Comment (Alphasaurs) [reply docs/querying/api.md]: removed

**Review by LeviHarrison — APPROVED** [review]
Great! Thank you so much, this has been needing to get done for a while.

**Pull request merged**